### PR TITLE
WD-2833 - Sort action logs

### PR DESCRIPTION
**Pull request by huwshimi**
## Done

- Sort action logs by default and add sorting to other columns.

## QA

- Pull code
- Run `dotrun clean && dotrun serve`
- Open http://0.0.0.0:8036/
- If you don't have any action logs go to a model -> click on an app -> tick some units and click the run action button.
- Go to the model's action logs tab.
- Check that the most recent logs are first.
- Check that you can sort the table using the headers.

## Details

Fixes: #1033.
https://warthogs.atlassian.net/browse/WD-2833

## Screenshots

Screenshot 2023-06-01 at 9 36 00 am

### CODE REVIEW

**Comment by huwshimi** [comment]
This is waiting for https://github.com/canonical/react-components/pull/918

**Comment by vladimir-cucu** [comment]
I'm still reviewing the code, but I've managed to find a few minor issues while QA-ing the changes:

1. Every 30sec there is an api call that resets any custom ordering in the table.
2. If there is no custom sorting (no up/down pointing arrow), then the header height gets slightly smaller.
3. Current issue appears when there is less horizontal space for a header cell than the length of the text within the header cell. Once we try to sort in ascending/descending order, the up/down pointing arrow will appear only if we click exactly on the place it should have been situated at. If we do not click on that place (for instance, we click on the text instead), the up/down arrow won't show up.

File: src/pages/EntityDetails/Model/ActionLogs/ActionLogs.tsx [hunk 1]
```diff
@@ -401,6 +433,9 @@ export default function ActionLogs() {
           emptyMsg={emptyMsg}
           columns={columnData}
           data={tableData}
+          initialSortColumn="completed"
+          initialSortDirection="descending"
```
[the remark this answers is not in the document]
Comment (huwshimi) [reply src/pages/EntityDetails/Model/ActionLogs/ActionLogs.tsx]: Done. I discovered there's a `sortInverted: true,` option I could set on the time column.

File: src/pages/EntityDetails/Model/ActionLogs/ActionLogs.tsx [hunk 2]
```diff
@@ -190,18 +255,22 @@ export default function ActionLogs() {
```
[the remark this answers is not in the document]
Comment (huwshimi) [reply src/pages/EntityDetails/Model/ActionLogs/ActionLogs.tsx]: Done!

**Comment by huwshimi** [comment]
> I'm still reviewing the code, but I've managed to find a few minor issues while QA-ing the changes:
> 
> 1. Every 30sec there is an api call that resets any custom ordering in the table.

This turns out to be a "feature" of React Table and I've filed an issue to fix this in ModularTable: https://github.com/canonical/react-components/issues/947.

> 2. If there is no custom sorting (no up/down pointing arrow), then the header height gets slightly smaller.

I've adjusted the column widths to improve the truncation on larger screens.

> 3. Current issue appears when there is less horizontal space for a header cell than the length of the text within the header cell. Once we try to sort in ascending/descending order, the up/down pointing arrow will appear only if we click exactly on the place it should have been situated at. If we do not click on that place (for instance, we click on the text instead), the up/down arrow won't show up.

This turned out to be an issue where the empty header in the controls column was displaying sort controls: https://github.com/canonical/react-components/issues/946.

**Review by vladimir-cucu — APPROVED** [review]

**Pull request merged**